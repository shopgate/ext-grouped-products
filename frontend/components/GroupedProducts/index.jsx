import React from 'react';
import PropTypes from 'prop-types';
import connect from './connector';
import styles from './style';

/**
 * The GroupedProducts component.
 * @param {props} param The component props.
 * @return {JSX}
 */
const GroupedProducts = ({ products }) => {
  if (!products.length) {
    return null;
  }

  return (
    <ul className={styles.container}>
      {products.map(({ name, id }) => (
        <li key={id}>{name}</li>
      ))}
    </ul>
  );
};

GroupedProducts.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default connect(GroupedProducts);
