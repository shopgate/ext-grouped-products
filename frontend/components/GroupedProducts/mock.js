import set from 'lodash/set';
import get from 'lodash/get';
import cloneDeep from 'lodash/cloneDeep';
import merge from 'lodash/merge';

const availability = {
  text: 'Lorem Ipsum',
  state: 'ok',
};

const stock = {
  ignoreQuantity: false,
  quantity: 30,
  info: 'Info',
  orderable: true,
  minOrderQuantity: 1,
  maxOrderQuantity: 1000,
};

const price = {
  currency: 'EUR',
  unitPrice: 5.0,
  unitPriceMin: 0,
  unitPriceStriked: 0,
  msrp: 0,
  discount: 0,
};

const product = {
  id: '1337',
  name: 'A shiny product',
  featuredImageUrl: 'https://cdn.acme.com/kitten.jpg',
  availability,
  price,
  stock,
};

let discounted = cloneDeep({
  ...product,
  id: '4711',
  name: 'A discounted product',
});

discounted = set(discounted, ['price', 'msrp'], 10.0);
discounted = set(discounted, ['price', 'discount'], 50);

let notOrderable = cloneDeep({
  ...product,
  id: '1234',
  name: 'A not orderable product',
});

notOrderable = set(notOrderable, ['stock', 'orderable'], false);
notOrderable = set(notOrderable, ['stock', 'quantity'], 0);

const mockedState = {
  product: {
    currentProduct: {
      productId: '1337',
    },
    productsById: {
      1337: {
        productData: product,
      },
    },
  },
};

export {
  mockedState,
  product as mockedProduct,
  discounted as mockedDiscountedProduct,
  notOrderable as mockedNotOrderableProduct,
};
