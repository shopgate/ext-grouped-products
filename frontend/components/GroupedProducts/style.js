import { css } from 'glamor';

const container = css({
  border: '2px solid red',
});

export default {
  container,
};
