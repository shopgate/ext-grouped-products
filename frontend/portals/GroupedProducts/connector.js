import { connect } from 'react-redux';
import { getGroupedProducts } from '../../selectors';

/**
 * Maps state to props.
 * @param {Object} state State.
 * @returns {Object}
 */
const mapStateToProps = state => ({
  products: getGroupedProducts(state),
});

export default connect(mapStateToProps);
