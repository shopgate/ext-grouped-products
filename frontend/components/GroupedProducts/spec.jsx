import React from 'react';
import { shallow } from 'enzyme';
import { hasFavorites, renderFlatButtons } from '../../helpers/index';
import GroupedProducts from './index';

jest.mock('../../helpers/index', () => ({
  renderFlatButtons: jest.fn(),
  hasFavorites: jest.fn(),
}));

const withProducts = [
  { id: '1234' },
  { id: '1337' },
];
const withoutProducts = [];

describe('<GroupedProducts />', () => {
  it('should not render if there is no products', () => {
    const wrapper = shallow(<GroupedProducts
      products={withoutProducts}
      renderFlatButtons={renderFlatButtons}
      hasFavorites={hasFavorites}
    />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').exists()).toBeFalsy();
  });

  it('should render if there are products', () => {
    const wrapper = shallow(<GroupedProducts
      products={withProducts}
      renderFlatButtons={renderFlatButtons}
      hasFavorites={hasFavorites}
    />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('List').exists()).toBeTruthy();
  });
});
