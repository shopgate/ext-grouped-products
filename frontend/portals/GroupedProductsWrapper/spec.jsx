import React from 'react';
import { shallow } from 'enzyme';
import GroupedProducts from './index';

jest.mock('../../components/GroupedProducts', () => props => <div {...props}>Hello world</div>);

describe('<GroupedProductsWrapper />', () => {
  it('should render', () => {
    const wrapper = shallow(<GroupedProducts />);
    expect(wrapper).toMatchSnapshot();
  });
});
