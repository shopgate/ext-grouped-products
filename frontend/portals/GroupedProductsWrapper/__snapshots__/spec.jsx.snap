// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupedProductsWrapper /> should render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
